pkg/restore/db.go

```diff
@@ -159,11 +160,23 @@ func (db *DB) CreateTable(ctx context.Context, table *utils.Table) error {
 		default:
 			alterAutoIncIDFormat = "alter table %s.%s auto_increment = %d;"
 		}
+
+		var autoIncID uint64
+		// auto inc id overflow
+		if table.Info.AutoIncID < 0 {
+			if table.Info.IsAutoIncColUnsigned() {
+				autoIncID = math.MaxUint64
```

Comment:
this still doesn't look correct. referring to the example before

```sql
create table a (id bigint unsigned primary key auto_increment);
insert into a values (12345678901234567890);
```

the auto_inc_id in the backupmeta is `-6101065172474953725` (12345678901234597891 &minus; 2<sup>64</sup>). 

when we restore we should use `12345678901234597891`, not `18446744073709551615` (MaxUint64).

Reply:
yes, you are right, when I try to fix this, I found that TiDB didn't handle unsigned auto_increment correctly, I file an issue to trace it https://github.com/pingcap/tidb/issues/19238

Reply:
do we need to wait for pingcap/tidb#19241? simply hard-casting as `uint64(table.Info.AutoIncID)` seems to be sufficient for AUTO_INCREMENT_ID (not sure about SEQUENCE 🤔)

Reply:
@3pointer pingcap/tidb#19241 has been merged